content/ytt/docs/latest/how-it-works.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "How it works"
+---
+
+## Overview
+
+Let's get an idea of how `ytt` works by looking at the high-level concepts and flow.
+
+## The Pipeline
+
+When you invoke `ytt` ...
+
+```console
+$ ytt -f (input files)
+```
+
+... you can think of it as a pipeline in four stages, looking something like this:
+
+![ytt pipeline overview](/images/ytt/ytt-pipeline-overview.jpg)
+<!-- source: https://miro.com/app/board/o9J_lIfcKZY=/?moveToWidget=3074457357774380591&cot=14 --> 
+
+where ...
+- each "input file" can have one or more YAML documents in them (separated by `---`). \
+  _(from here on out, we'll refer to YAML documents as just "documents".)_
+- if a document contains one or more `ytt` annotations (i.e. lines that start with `#@`), it's called a "template".
+  ```yaml
+  # No annotations to be seen; just a plain old YAML document, here.
+  ---
+  foo: 14
+  bar:
+  - Hello, Alice
+  - Hello, Bob
+  - Hello, world
+  ```
+  vs.
+  ```yaml
+  #! This is a ytt template.
+  
+  ---
+  foo: 14
+  bar:
+  #@ for/end name in ["Alice", "Bob", "world"]:
+  - #@ "Hello, " + name
+  ```
+- when a template starts with the [`@data/values`](ytt-data-values.md#declaring-and-using-data-values) annotation, it's called a "Data Values":
```

Comment:
shouldnt this say YAML document instead of template?

Reply:
Yeah.... so there's this needle I'm trying to thread through this string of terms in hopes of making meaningful distinctions between them... and because the word "template" is overloaded, it complicates the story:

"template" can have three (3) meanings:
1. a file containing `ytt` directives
2. a document containing `ytt` directives
3. (with a capital 'T', "Template") a document that's contributing to output (i.e. _not_ a Data Values nor an Overlay)

And the naming of parts goes something like this:
- a file contains a document set
- a file containing annotations is a template (definition 1; not used in this story)
- a document set contains zero or more documents
- a document, if containing annotations is a template (definition 2)
- templates can be categorized into three groups:
  - Data Values
  - Overlays
  - Templates (definition 3)

I've been trying to avoid getting wordy. For example, could use _these_ terms:

1. a file containing `ytt` directives is a "template"
2. a document containing `ytt` directives is a "templated document"
3. a "templated document" contributing to output is a "templated output document"

Folding that back in to the sentence at hand:

```suggestion
- when a templated document starts with the [`@data/values`](ytt-data-values.md#declaring-and-using-data-values) annotation, it's called a "Data Values":
```

It's a little wordier, but maybe the increased density isn't too much? ... and it has the added benefit of foreshadowsingterms that have meaning (e.g. marking a file for output makes it one of those "output documents")...